websurvey/src/routes/Form.svelte

```diff
@@ -22,17 +24,22 @@
     $: {
         index = form.fields.map(({ ref }) => ref).indexOf(ref);
         field = form.fields[index];
+        required = field.validations.required;
     }
 
     const responseStore = new ResponseStore();
 
     const handleSubmit = () => {
-        let snapshot = responseStore.snapshot(ref, fieldValue);
-        let qa = responseStore.getQa(snapshot);
-        if (index < form.fields.length - 1) {
+        const snapshot = responseStore.snapshot(ref, fieldValue);
+        const qa = responseStore.getQa(snapshot);
+        const isValid = validateFieldValue(field, fieldValue, required);
```

Comment:
Hrm, we can go to the thankyou screen if the last answer was not valid? What happens if it's not valid? 

This is starting to look great!

Can you push this idea even further and extract some more of this logic out of the component? How dumb can you make this component?